src/geos.c

```diff
@@ -17,6 +20,11 @@ int init_geos(PyObject* m) {
   geos_exception[0] = PyErr_NewException("shapely.errors.GEOSException", base_class, NULL);
   PyModule_AddObject(m, "GEOSException", geos_exception[0]);
   return 0;
+
+  void* context_handle = GEOS_init_r();
+  // GEOSContext_setErrorMessageHandler_r(context_handle, geos_error_handler, last_error);
```

Comment:
Is there a reason not to hook up the error handler to the global context handle?  If so, a comment would be helpful to indicate why.

Reply:
Sorry, I should have added an inline comment here on github to ask about that.

So the problem is that this needs a `last_error`, a variable that is otherwise set up in `GEOS_INIT`. Should we have a "global" `last_error` variable as well? But then we also need another helper to check for the error (the current macro is set up to work with GEOS_INIT/GEOS_FINISH). 

Currently this global context is only use in `GeometryObject_dealloc`, where we actually don't do any error handling at the moment. But if we want to use this global context in other places as well (eg repr, hashing), then we will certainly need to set up the error handler.

Reply:
Thanks for the explanation.  I think a comment indicating it is intentionally not added right now since global context handle is only used where error handling is not used.  Maybe also a TODO?  Adding more macros can happen as part of a later PR if / when integrating the global context handle into more places.